tree/tree/src/TTreeCache.cxx

```diff
@@ -1266,6 +1266,8 @@ Bool_t TTreeCache::FillBuffer()
    if (showMore || gDebug > 6)
       Info("FillBuffer", "Looking at cluster spanning from %lld to %lld", fEntryCurrent, fEntryNext);
 
+   if (0 <= fNextClusterStart && fEntryCurrent < fNextClusterStart && !(fEntryCurrent < fNextClusterStart))
```

Comment:
`fEntryCurrent < fNextClusterStart && !(fEntryCurrent < fNextClusterStart)` doesn't seem to make sense.

Reply:
You are right.  This was intended to detect when the user is moving backward but a "typo/thinko" mess it up :( rendering the whole PR moot ... i.e. Test need to go in at the same time...